front/lib/api/assistant/messages.ts

```diff
@@ -237,20 +211,30 @@ export async function batchRenderAgentMessages(
     const agentMessage = message.agentMessage;
 
     let action: AgentActionType | null = null;
-    if (retrievalActionByAgentMessageId[agentMessage.id]) {
-      const actionId = retrievalActionByAgentMessageId[agentMessage.id].id;
-      action = agentRetrievalActions.find((a) => a.id === actionId) || null;
-    } else if (agentMessage.agentDustAppRunActionId) {
+
+    // TODO: Allow multiple actions per agent messages.
+    if (
+      agentRetrievalActions.filter((a) => a.agentMessageId === agentMessage.id)
```

Comment:
We probably don't need to go through every action twice right ? Could just have the `find` call ?

Reply:
really nit though, I guess we don't really care